File: lab3/validate.py
#!/usr/bin/python

import sys
import re

def chkname(x, y):
    if re.search(r"^[A-Z]([a-z]|[A-Z]|-)+$", y):
        return(x, y)
    else:
        nerror(x)

def chkzip(x, y):
    if re.search(r"^(\d){5}$", y):
        return(x, y)
    else:
        print x,"must be exactly 5 digits!"
        sys.exit()

def chkadd(x, y):
    if re.search(r"^(\d|\w|\.|-)+@(\d|\w|\.|-)+.(\d|\w|\.|-)+$", y):
        print "Validated!"
        return(x, y)
    else:
        print x,"must be USER\@DOMAIN, where both USER and DOMAIN must be only letters, numbers, dots, underscores, and hyphens!"
        sys.exit()

def nerror(x):
    print x, "must start with a captial letter and contain only letters and hypens!"
    sys.exit()



if len(sys.argv) != 1:
    print "usage: ./validate.py"
else:
    input = raw_input("First name: ")
    chkname("First name", input)

    input = raw_input("Last name: ")
    chkname("Last name", input)
    
    input = raw_input("Zip code: ")
    chkzip("Zip Code", input)

    input = raw_input("Email Address: ")
    chkadd("Email Address", input)

File: lab2/validate.pl
#!/usr/bin/perl
#validate

use strict;
use warnings;

my $input;
if (@ARGV == 0)
{
	print "First name: ";
	$input = <STDIN>;
	&chkname("Firs name", $input);

	print "Last name: ";
	$input = <STDIN>;
	&chkname("Last name", $input);

	print "Zip code: ";
	$input = <STDIN>;
	&chkzip("Zip code", $input);
	
	print "Email Address:";
	$input = <STDIN>;
	&chkadd("Email Address", $input);
}

else
{
	die "usage: ./validate.pl \n";
}


sub chkname {
	if ($_[1] =~ /^[A-Z]([a-z]|[A-Z]|-)+$/ ){
		#&out;
        }
	else{
		&nerror("$_[0]");
	}
}

sub chkzip {
	if ($_[1] =~ /^(\d){5}$/){
		#&out;	
	}	
	else{
		die "$_[0] must be exactly 5 digits!\n";
	}
}

sub chkadd {
	if ($_[1] =~ /^(\d|\w|\.|-)+@(\d|\w|\.|-)+.(\d|\w|\.|-)+$/){
		#&out;
		print "Valiated!\n";
	}
	else{
		print "$_[0] must be USER\@DOMAIN, where both USER and DOMAIN must be only letters, numbers, dots, underscores, and hyphens!\n";
	}
}

sub nerror {
        die "$_[0] must start with a capital letter and contain only letters and hypens!\n";
}

#sub out{
#       print "$_[0]: $_[1]";
#}
